src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -615,13 +648,13 @@ Visitor::SubQueryAndMaybeValues Visitor::visitTypesafe(
 }
 
 // ____________________________________________________________________________________
-GroupKey Visitor::visitTypesafe(Parser::GroupConditionContext* ctx) {
+GroupKey Visitor::visit(Parser::GroupConditionContext* ctx) {
   // TODO<qup42> Deploy an abstraction `visitExpressionPimpl(someContext*)` that
   // performs exactly those two steps and is also used in all the other places
   // where we currently call
-  // `SparqlExpressionPimpl(visitTypesafe(ctx->something()))` manually.
+  // `SparqlExpressionPimpl(visit(ctx->something()))` manually.
```

Comment:
Would now be a good place to deploy this abstraction?

Below I saw a `SparqlExpressionPimpl` that doesn't have its descriptor set. 
I suggest enforcing this by making a two-argument constructor to the Pimpl class that enforces the setting of the descriptor.

Reply:
I have now implemented this in two separate steps. The new constructor and a helper that uses this constructor. This way separation of concerns is preserved while also having short code.